src/org/traccar/helper/Parser.java

```diff
@@ -165,4 +167,122 @@ public double nextCoordinate() {
         return nextCoordinate(CoordinateFormat.DEG_MIN_HEM);
     }
 
+    public enum DateTimeFormat {
+        HMS,         // HHMMSS
+        SMH,         // SSMMHH
+
+        HMS_YMD,     // HHMMSSYYYYMMDD or HHMMSSYYMMDD
+        HMS_DMY,     // HHMMSSDDMMYYYY or HHMMSSDDMMYY
+        SMH_YMD,     // SSMMHHYYYYMMDD or SSMMHHYYMMDD
+        SMH_DMY,     // SSMMHHDDMMYYYY or SSMMHHDDMMYY
+
+        DMY_HMS,     // DDMMYYYYHHMMSS or DDMMYYHHMMSS
+        DMY_HMSms,   // DDMMYYYYHHMMSS.sss or DDMMYYHHMMSS.sss
+        YMD_HMS,     // YYYYMMDDHHMMSS or YYMMDDHHMMSS
+        YMD_HMSms,   // YYYYMMDDHHMMSS.sss or YYMMDDHHMMSS.sss
+
+        ISO8601      // YYYY-MM-DDTHH:MM:SS+HH:MM
```

Comment:
This is never used. Please remove. I think for ISO parsing we can just use standard Java classes.